src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -2655,7 +2656,7 @@ static CompositeChange clearTransportedByForAlliedAirOnCarrier(final Collection<
     if (!carriers.isEmpty() && !games.strategy.triplea.Properties.getAlliedAirIndependent(data)) {
       final Match<Unit> alliedFighters = new CompositeMatchAnd<>(Matches.isUnitAllied(attacker, data),
           Matches.unitIsOwnedBy(attacker).invert(), Matches.UnitIsAir, Matches.UnitCanLandOnCarrier);
-      final Collection<Unit> alliedAirInTerr = Match.getMatches(battleSite.getUnits().getUnits(), alliedFighters);
+      final Collection<Unit> alliedAirInTerr = Match.getMatches(attackingUnits, alliedFighters);
```

Comment:
@veqryn Yeah, I checked that and only the attacking units are checked to see if they should have transportedBy set before the battle not defenders. Which logically makes sense since allied defending air units will always defend their carriers.

IMO, using transportedBy for allied fighters on carriers is kind of just a hack to have them not participate as attackers and be killed off if the carrier dies. It appears to only be used for carriers right before battles to set the flag and then after battles to clear it. Carriers/air units should probably have their own system for tracking which air units are on which carriers explicitly. This could then just be checked in battle logic so the transportedBy would only be used by transports. The transport system could also use an overhaul especially regarding land/air transports. But doing either of those things would be considerable effort :)